File: Backend/models/ProviderApplication.js
import mongoose from "mongoose";

const providerApplicationSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    businessName: {
      type: String,
      required: true,
      trim: true,
    },
    businessDescription: {
      type: String,
      required: true,
    },
    category: {
      type: String,
      required: true,
    },
    experience: {
      type: String,
      required: true,
    },
    phone: {
      type: String,
      required: true,
    },
    address: {
      type: String,
      required: true,
    },
    certifications: {
      type: String,
    },
    portfolio: {
      type: String,
    },
    businessImage: {
      type: String,
      default: null,
    },
    status: {
      type: String,
      enum: ["pending", "approved", "rejected"],
      default: "pending",
    },
    adminNotes: {
      type: String,
    },
    reviewedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
    reviewedAt: {
      type: Date,
    },
    submittedAt: {
      type: Date,
      default: Date.now,
    },
  },
  {
    timestamps: true,
  },
);

export default mongoose.model("ProviderApplication", providerApplicationSchema);

File: Backend-SpringBoot/src/main/java/com/servesync/model/ProviderApplication.java
package com.servesync.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "provider_applications")
public class ProviderApplication {
    
    @Id
    private String id;
    
    @DBRef
    private User user;
    
    private String userId;
    
    private String businessName;
    
    private String description;
    
    private String category;
    
    private Integer experience;
    
    private List<String> certifications;
    
    private String documentsUrl;
    
    private ApplicationStatus status = ApplicationStatus.PENDING;
    
    private String adminNotes;
    
    private String rejectionReason;
    
    @CreatedDate
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    private LocalDateTime updatedAt;
    
    public enum ApplicationStatus {
        PENDING, APPROVED, REJECTED;
        
        @com.fasterxml.jackson.annotation.JsonCreator
        public static ApplicationStatus fromString(String value) {
            return ApplicationStatus.valueOf(value.toUpperCase());
        }
    }
}
